lib/shrine/plugins/derivation_endpoint.rb

```diff
@@ -0,0 +1,1295 @@
+    # inline in the browser, while other derivatives will be force downloaded.
+    #
+    #     plugin :derivation_endpoint, disposition: -> {
+    #       name == :thumbnail ? "inline" : "attachment"
+    #     }
+    #
+    # ## Host
+    #
+    # By default generated URLs are relative. To generate absolute URLs, you
+    # can pass the `:host` option:
+    #
+    #     plugin :derivation_endpoint, host: "https://example.com"
+    #
+    # Now the generated URLs will include the specified URL host:
+    #
+    #     uploaded_file.derivation_url(:thumbnail)
+    #     #=> "https://example.com/thumbnail/eyJpZCI6ImZvbyIsInN?signature=..."
+    #
+    # You can also pass `:host` per URL:
+    #
+    #     uploaded_file.derivation_url(:thumbnail, host: "https://example.com")
+    #     #=> "https://example.com/thumbnail/eyJpZCI6ImZvbyIsInN?signature=..."
+    #
+    # ## Prefix
+    #
+    # If you're mounting the derivation endpoint under a path prefix, the
+    # `:prefix` option needs to match in order for correct URLs to be generated:
+    #
+    #     plugin :derivation_endpoint, prefix: "derivations/image"
+    #
+    # Now generated URLs will include the specified path prefix:
+    #
+    #     uploaded_file.derivation_url(:thumbnail)
+    #     #=> "/derivations/image/thumbnail/eyJpZCI6ImZvbyIsInN?signature=..."
+    #
+    # You can also pass `:prefix` per URL:
+    #
+    #     uploaded_file.derivation_url(:thumbnail, prefix: "derivations/image")
+    #     #=> "/derivations/image/thumbnail/eyJpZCI6ImZvbyIsInN?signature=..."
+    #
+    # ## Expiration
+    #
+    # By default generated URLs are valid indefinitely. If you want URLs to
+    # expire after a certain amount of time, you can set the `:expires_in`
+    # option:
+    #
+    #     plugin :derivation_endpoint, expires_in: 90
+    #
+    # Now any URL will stop being valid 90 seconds after it was generated:
+    #
+    #     uploaded_file.derivation_url(:thumbnail)
+    #     #=> "/thumbnail/eyJpZCI6ImZvbyIsInN?expires_at=1547843568&signature=..."
+    #
+    # You can also pass `:expires_in` per URL:
+    #
+    #     uploaded_file.derivation_url(:thumbnail, expires_in: 90)
+    #     #=> "/thumbnail/eyJpZCI6ImZvbyIsInN?expires_at=1547843568&signature=..."
+    #
+    # ## Response headers
+    #
+    # ### Content Type
+    #
+    # By default in the derivation response, the [`Content-Type`] header is
+    # inferred from the file extension of the derivative (using `Rack::Mime`).
+    # You can override that with the `:type` option:
+    #
+    #     plugin :derivation_endpoint, type: -> {
+    #       "image/webp" if name == :webp
+    #     }
+    #
+    # If the block returns `nil`, then the default type will be set. You can
+    # also set `:type` per URL:
+    #
+    #     uploaded_file.derivation_url(:webp, type: "image/webp")
+    #     #=> "/webp/eyJpZCI6ImZvbyIsInN?type=image%2Fwebp&signature=..."
```

Comment:
None of the other examples use a prefix, so that you can see the relevant parts of the URL more clearly. But if it's confusing I should probably add it back.